src/vm/internal/otable_test.cc

```diff
@@ -0,0 +1,48 @@
+#define TESTING
+
+#include<string>
+struct object_t
```

Comment:
I can tell you how to make this work without declaring a fake object_t() here.

Declaring a function in otable.c 
```
std::string obname(object_t* x) {
return std::string(x->obname);
}
```
and use it here. This way I think otable_test.cc don't need to know about struct object_t .

Reply:
The problem is otable.cc needs the fake object_t at present because I don't want the dependencies from all the forward declarations in object.h. I think there might be a way to resolve this by creating a stripped down object.h with the fake object_t and changing the include path. But do we need to think about this issue carefully now?  I think mock objects might generall pose problems for C++. See this-

https://stackoverflow.com/questions/5426091/does-mock-objects-in-c-always-requires-virtual-methods-or-templates

I think perhaps the sneaky way of solving the mock problem is to instead of having an abstract parent class and using some sort of dependency injection is to rely on the linker. But I would like to hear your opinion.

Reply:
Actually I will try doing this the normal way. Concerning the verbose code your suggestion actually introduces a but you need to do both verbose and !verbose test since the operator is on a signed integer actually the system passes -1 to it.